src/Hl7.Fhir.ElementModel/TypedElementOnSourceNode.cs

```diff
@@ -166,6 +166,21 @@ private string deriveInstanceType(ISourceNode current, IElementDefinitionSummary
                         raiseTypeError($"Choice element '{current.Name}' is suffixed with unexpected type '{suffix}'", current, location: current.Location);
                 }
             }
+            else if (info.Representation == XmlRepresentation.TypeAttr) // May be used by models other then FHIR, e.g. CCDA represented by a StructureDefinition
+            {
+                if(info.Type.Count() == 1)
+                {
+                    instanceType = info.Type.Single().GetTypeName();
+                }
+                else
+                {
+                    var typeName = current.Children("type").FirstOrDefault()?.Text;
+                    if (typeName != null)
+                        instanceType = info.Type.Where(type => type.GetTypeName().EndsWith(typeName)).FirstOrDefault().GetTypeName();
```

Comment:
Unfortunately this will fail. For example when the instance has the type `PQ` then the StructureDefinition has 3 possible candidates: `PQ`, `IVL_PQ` and `RTO_PQ_PQ` when we use the method `EndsWith()`. Incidentally it takes the first so we are safe, but it could wrong if the order is different.

Reply:
So you need something like this:

```
public static string Last(this string str) 
{
   var pos = str.LastIndexOf('/');
   return pos > -1 ? str.Substring(pos + 1) : str;
}

```
and then
```
instanceType = info.Type.Where(type => type.GetTypeName().Last() == typeName).FirstOrDefault().GetTypeName();
```

Reply:
Changed like requested, 'Last' was a tricky name to choose as it's already in use.